src/prefect/blocks/core.py

```diff
@@ -100,6 +100,37 @@ def _get_non_block_reference_definitions(object_definition: Dict, definitions: D
     return non_block_definitions
 
 
+def _is_subclass(cls, parent_cls) -> bool:
+    """
+    Checks if a given class is a subclass of another class.
+    """
+    return inspect.isclass(cls) and issubclass(cls, parent_cls)
+
+
+def _collect_secret_fields(name: str, type_: Type, secrets: List[str]) -> None:
+    """
+    Collects all secret fields from a given type.
```

Comment:
Can you expand this a little bit? Mainly would be helpful to note it's recursive with support for unions/base models/block types?